clang/lib/CodeGen/CGCUDARuntime.h

```diff
@@ -63,6 +63,12 @@ class CGCUDARuntime {
       OffloadGlobalSurfaceEntry = 0x2,
       /// Mark the entry as a texture variable.
       OffloadGlobalTextureEntry = 0x3,
+      /// Mark the entry as being extern.
+      OffloadGlobalExtern = 0x4,
```

Comment:
should we start from 0x8 to reserve some entries for future variable types?

Reply:
They could go out of order in any case, though I'm halfway considering reworking this struct format in the future.

Reply:
once people start to use these enums, it is difficult to change them since it will break binary backward compatibility